Zend/tests/clone_initializer/clone_initializer_success2.phpt

```diff
@@ -0,0 +1,31 @@
+--TEST--
+Test that dynamic properties can be initialized during cloning
+--FILE--
+<?php
+
+class Foo
+{
+    public function withProperties()
+    {
+        return clone $this with {
+            property1: 1,
+            property2: "foo",
```

Comment:
Could add a test of this with function calls, e.g.
- If a previous property expression throws, subsequent ones are not evaluated?
- If an expression throws, type checks are not performed (e.g. `clone $this with { strTypedProperty: throw new Exception("test") }`)
- temporary values such as `strtolower("ABC")` are freed if there's a type error assigning to a typed property

Reply:
Thanks for the ideas, I'll definitely add such tests!